WooCommerce/WooCommerceTests/ViewRelated/Reviews/ReviewsViewControllerTests.swift

```diff
@@ -0,0 +1,101 @@
+import XCTest
+@testable import WooCommerce
+
+/// Tests for `ReviewsViewController`.
+///
+final class ReviewsViewControllerTests: XCTestCase {
+    private var mockViewModel: MockReviewsViewModel!
+    private var sut: ReviewsViewController!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        mockViewModel = MockReviewsViewModel(siteID: 123)
+        sut = ReviewsViewController(viewModel: mockViewModel)
+    }
+
+    override func tearDownWithError() throws {
+        mockViewModel = nil
+        sut = nil
+
+        try super.tearDownWithError()
+    }
+
+    func test_menu_bar_button_item_is_not_present_if_there_are_no_unread_notifications() {
+        // When
+        mockViewModel.hasUnreadNotifications = false
+        sut.makeViewAppear()
+
+        // Then
+        XCTAssertNil(sut.navigationItem.rightBarButtonItem)
+    }
+
+    func test_menu_bar_button_item_is_visible_if_there_are_unread_notifications_available() throws {
+        // When
+        mockViewModel.hasUnreadNotifications = true
+        sut.makeViewAppear()
+
+        // Then
+        let markAllAsReadyButton = try XCTUnwrap(sut.navigationItem.rightBarButtonItem)
+        XCTAssertEqual(markAllAsReadyButton.accessibilityIdentifier, "reviews-open-menu-button")
+    }
+}
+
+// MARK: ReviewsViewController helpers
+//
+private extension ReviewsViewController {
+    func makeViewAppear() {
+        loadViewIfNeeded()
+        beginAppearanceTransition(true, animated: false)
+        endAppearanceTransition()
```

Comment:
It seems to me that these two method calls aren't really needed, since the tests pass anyway without them. Do you think they can be removed?

Reply:
Removing these methods will end up in `ReviewsViewControllerTests` executing only `viewDidLoad` method of `ReviewsViewController`. `viewWillAppear` method will not be executed.

I believe that it will make the tests incomplete.

Reply:
Thank you! That makes sense, but do you think we need `viewWillAppear` only if we need to test its functionality? Either way, I think it's fine to keep it as is.

Reply:
I think if we are testing for something that can change in `viewWillAppear` IMO it is safer to let the view controller appear instead of only loading the view.
But, I agree with your point. There are many scenarios in which this can be unnecessary. 

I can remove this if you prefer that.

Reply:
Let's keep it for now 👍